packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -152,13 +152,15 @@ const findIntersectionAndUpdateTxStore = ({
           lowerBound !== undefined && `since block ${lowerBound}`
         );
 
-        const txRolledBack = lastStoredTransaction && newTransactions[0]?.id !== lastStoredTransaction.id;
+        const txRolledBack =
+          lastStoredTransaction && newTransactions[newTransactions.length - 1]?.id !== lastStoredTransaction.id;
```

Comment:
I think this should be looking only at the first block from `newTransactions`
```diff
- lastStoredTransaction && newTransactions[newTransactions.length - 1]?.id !== lastStoredTransaction.id;
+ const newTxFirstBlock = newTransactions.filter(tx => tx.blockHeader.blockNo === lowerBound);
+ lastStoredTransaction && newTxFirstBlock[newTxFirstBlock.length - 1]?.id !== lastStoredTransaction.id;
```

Reply:
I am rewriting this logic here, sec